src/utils/error.ts

```diff
@@ -0,0 +1,21 @@
+import { cast_u64_ptr } from "../base_utils/convert.ts";
+import { require } from "../gi.js";
+import { ExtendedDataView } from "./dataview.js";
+
+export function createGError(errorBuffer: ArrayBuffer) {
+  const GError = getGLibError();
+
+  const dataView = new ExtendedDataView(errorBuffer);
+  const pointer = cast_u64_ptr(dataView.getBigUint64());
+
+  const error = Object.create(GError.prototype);
+  Reflect.defineMetadata("gi:ref", pointer, error);
+
+  error.stack = new Error().stack;
+
+  return error;
+}
+
+export function getGLibError() {
+  return require("GLib", "2.0").Error;
```

Comment:
I think it's not a good idea to use introspection inside deno_gi as it can lead to recursive dependency on it's self.
is it possible to do this with only using static bindings?

Reply:
I don't think it's the best idea ever, but it's the best approach.

This is because the `GLib.Error` must be consistent with other instances of `GLib.Error` so that we can do instance checks like so:

```js
try {
  const file = Glib.file_get_contents("/path/to");
} catch (err) {
  if (err instanceof GLib.Error && error.matches(GLib.FileError.NOENT)) {
	console.error("The error was not found");
  } else {
     //... handle error differently
  }
}
```

The above code would not work otherwise because `GLib.Error` would be a completely different instance from `error`.

Reply:
okay that makes sense